src/Microsoft.ML.Data/DataLoadSave/DataLoadSaveCatalog.cs

```diff
@@ -2,19 +2,92 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Transforms;
+
 namespace Microsoft.ML.Runtime
 {
     /// <summary>
-    /// A catalog of operations to load and save data.
+    /// A catalog of operations over data that are not transformers or estimators.
+    /// This includes data readers, saving, caching, filtering etc.
     /// </summary>
-    public sealed class DataLoadSaveOperations
+    public sealed class DataOperations
     {
         internal IHostEnvironment Environment { get; }
 
-        internal DataLoadSaveOperations(IHostEnvironment env)
+        internal DataOperations(IHostEnvironment env)
         {
             Contracts.AssertValue(env);
             Environment = env;
         }
+
+        /// <summary>
+        /// Creates a lazy in-memory cache of <paramref name="input"/>.
+        /// Caching happens per-column. A column is only cached when it is first accessed.
+        /// In addition, <paramref name="columnsToPrefetch"/> are considered 'always needed', so all of them
+        /// will be cached whenever any data is requested.
+        /// </summary>
+        /// <param name="input">The data view to cache.</param>
+        /// <param name="columnsToPrefetch">The columns that must be cached whenever anything is cached. Empty array or null
+        /// is acceptable, it means that all columns are only cached at the first access.</param>
+        public IDataView Cache(IDataView input, params string[] columnsToPrefetch)
+        {
+            Environment.CheckValue(input, nameof(input));
+            Environment.CheckValueOrNull(columnsToPrefetch);
+
+            int[] prefetch = new int[Utils.Size(columnsToPrefetch)];
+            for (int i = 0; i < prefetch.Length; i++)
+            {
+                if (!input.Schema.TryGetColumnIndex(columnsToPrefetch[i], out prefetch[i]))
+                    throw Environment.ExceptSchemaMismatch(nameof(columnsToPrefetch), "prefetch", columnsToPrefetch[i]);
+            }
+            return new CacheDataView(Environment, input, prefetch);
+        }
+
+        /// <summary>
+        /// Keep only those rows that satisfy the range condition: the value of column <paramref name="columnName"/>
+        /// must be between <paramref name="minimum"/> and <paramref name="maximum"/>, inclusive.
+        /// </summary>
+        /// <param name="input">The input data</param>
+        /// <param name="columnName">The name of a column to use for filtering</param>
+        /// <param name="minimum">The inclusive lower bound</param>
+        /// <param name="maximum">The inclusive upper bound</param>
```

Comment:
>inclusive upper bound [](start = 38, length = 21)

How much trouble would it be to make it exclusive? The reason why I might like this is, let's imagine I want to split my dataset in half(ish) somehow. I generate a random float from 0 to 1. I can't however do [0.0, 0.5] and [0.5, 1.0] though, because things *equal* to 0.5 will appear in both. Also merely relying on the fact that a 0.5 value specifically is somewhat rare also makes me quite nervous.

Reply:
I don't like the idea of exclusive upper bound either, to be honest. 

---
In reply to: [231714820](https://github.com/dotnet/machinelearning/pull/1569#discussion_r231714820) [](ancestors = 231714820)

Reply:
Sorry, I meant no disrespect :) I actually meant to say that "I don't like the idea of **inclusive** upper bound either, to be honest. "


---
In reply to: [231728302](https://github.com/dotnet/machinelearning/pull/1569#discussion_r231728302) [](ancestors = 231728302)